src/python/ensembl/io/genomio/genome_metadata/compare_genome_stats.py

```diff
@@ -0,0 +1,212 @@
+
+    # NCBI includes the chromodomes in its stats
+    core_adjusted_scaffolds = core_num_scaffolds + core_num_chrs
+
+    # Compile the counts
+    ncbi_counts = {
+        "num_organella": len(ncbi_organella),
+        "num_chromosomes": ncbi_main.get("total_number_of_chromosomes", 0),
+        "num_scaffolds": ncbi_main.get("number_of_scaffolds", 0),
+        "num_contigs": ncbi_main.get("number_of_contigs", 0),
+    }
+    core_counts = {
+        "num_organella": core_num_organella,
+        "num_chromosomes": core_adjusted_chrs,
+        "num_scaffolds": core_adjusted_scaffolds,
+        "num_contigs": core["coord_system"].get("contig", 0),
+    }
+
+    # Only compare contigs if there are any in the core db
+    if ncbi_info.get("assembly_level") != "Contig":
+        del ncbi_counts["num_contigs"]
+        del core_counts["num_contigs"]
+
+    return _diff_dicts(ncbi_counts, core_counts)
+
+
+def compare_annotation(ncbi: Dict, core: Dict) -> Dict:
+    """Compare NCBI vs Core annotation stats (biotype counts).
+
+    Args:
+        ncbi: Dict of biotype counts from NCBI.
+        core: Dict of biotype counts from the core.
+
+    Returns:
+        Dict: Each count from NCBI, from Core, and their diff.
+    """
+    # Prepare counts to be comparable
+    core_biotypes = core.get("genes", {}).get("biotypes", {})
+
+    # We want a value for:
+    # protein_coding = same
+    # pseudogene = all pseudogene biotypes
+    # total = same
+    # other = number of misc_RNA
+
+    ncbi_counts = {
+        "protein_coding": ncbi.get("protein_coding", 0),
+        "pseudogene": ncbi.get("pseudogene", 0),
+        "total_genes": ncbi.get("total", 0),
+        "other": ncbi.get("other", 0),
+    }
+
+    # Add all pseudogenes
+    num_pseudogenes = 0
+    for name, num in core_biotypes.items():
+        if re.match(".*pseudogen.*", name):
+            num_pseudogenes += num
+
+    # Others? misc_mRNA (or anything similar?)
+    num_others = core_biotypes.get("misc_RNA", 0)
+
+    core_counts = {
+        "protein_coding": core_biotypes.get("protein_coding", 0),
+        "pseudogene": num_pseudogenes,
+        "total_genes": core.get("genes", {}).get("total", 0),
+        "other": num_others,
+    }
+
+    return _diff_dicts(ncbi_counts, core_counts)
+
+
+def compare_stats(ncbi: Dict, core: Dict) -> Dict[str, Any]:
+    """Compare stats from NCBI and our core.
+
+    Args:
+        ncbi: Dict of stats from NCBI.
+        core: Dict of stats from the core.
+
+    Returns:
+        Dict: Each count from NCBI, from Core, and their diff.
+    """
+
+    ncbi_annotation_stats = ncbi.get("annotation_info", {}).get("stats", {}).get("gene_counts", {})
+    core_assembly_stats = core.get("assembly_stats", {})
+    core_annotation_stats = core.get("annotation_stats", {})
+
+    comp: Dict[str, Any] = {
+        "assembly_diff": compare_assembly(ncbi, core_assembly_stats),
+    }
+    if core_annotation_stats is not None and ncbi_annotation_stats is not None:
```

Comment:
Based on the definitions above, neither `core_annotation_stats` nor `ncbi_annotation_stats` can be `None`, because the `compare_*` functions are expecting dictionaries, and their typing is not allowing `Optional`. So I'm afraid this check is not coherent with the previous code.